specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json

```diff
@@ -5724,6 +5803,62 @@
           "description": "The ip address of the next hop."
         }
       }
+    },
+    "EffectiveRouteList": {
+      "description": "EffectiveRoutes List.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EffectiveRoute"
+          },
+          "description": "The list of effective routes configured on the virtual hub or the specified resource."
+        }
+      }
+    },
+    "EffectiveRoute": {
+      "description": "The effective route.",
+      "properties": {
+        "addressPrefixes": {
+          "description": "The list of address prefixes.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "nextHops": {
+          "description": "The list of next hops.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "nextHopType": {
+          "description": "The type of the next hop.",
+          "type": "string"
+        },
+        "asPath": {
+          "description": "The ASPath of this route.",
+          "type": "string"
+        },
+        "routeOrigin": {
+          "description": "The origin of this route.",
+          "type": "string"
+        }
+      }
+    },
+    "EffectiveRoutesParameters": {
+      "description": "The parameters specifying the resource whose effective routes are being requested.",
+      "properties": {
+        "resourceId": {
+          "$ref": "./network.json#/definitions/Resource",
+          "description": "The resource."
+        },
+        "virtualWanResourceType": {
+          "type": "string",
+          "description": "The type of the specified resource like RouteTable, ExpressRouteConnection, HubVirtualNetworkConnection, VpnConnection and P2SConnection."
```

Comment:
I'm just curious. Why don't we use enum for this type?

Reply:
we dont want to restrict to certain values and would like this to be extendable in the future , hence nto using enum